packages/api/src/platforms/vtex/resolvers/query.ts

```diff
@@ -106,13 +108,34 @@ export const Query = {
 
     const after = maybeAfter ? Number(maybeAfter) : 0
     const searchArgs = {
-      page: Math.ceil(after / first!),
+      page: Math.ceil(after / first),
       count: first,
       query: term,
       sort: SORT_MAP[sort ?? 'score_desc'],
       selectedFacets: selectedFacets?.flatMap(transformSelectedFacet) ?? [],
     }
 
+    /**
+     * In case we are using crossSelling, we need to modify the search
+     * we will be performing on our search engine. The idea in here
+     * is to use the cross selling API for fetching the productIds our
+     * search will return for us.
+     * Doing this two request workflow makes it possible to have cross
+     * selling with Search features, like pagination, internationalization
+     * etc
+     */
+    const crossSelling = findCrossSelling(selectedFacets)
+    if (crossSelling) {
+      const products = await ctx.clients.commerce.catalog.products.crossselling({
+        type: FACET_CROSS_SELLING_MAP[crossSelling.key],
+        productId: crossSelling.value,
+      })
+
+      searchArgs.query = `product:${
```

Comment:
You can create something of the kind `let query = term` and move this bit of code up, only instead of rewriting `searchArgs.query`, you rewrite `query`. It is essentially the same thing, but the `let query` bit makes it obvious to the reader that this variable can (and probably will) be reassigned.